File: app/src/main/java/com/hillal/acc/data/local/dao/TransactionDao.java
package com.hillal.acc.data.local.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;
import androidx.room.OnConflictStrategy;
import com.hillal.acc.data.local.entity.Transaction;
import java.util.List;

@Dao
public interface TransactionDao {
    @Query("SELECT * FROM transactions")
    List<Transaction> getAllTransactions();

    @Query("SELECT * FROM transactions WHERE id = :id")
    Transaction getTransactionById(long id);

    @Query("SELECT * FROM transactions WHERE server_id = :serverId")
    Transaction getTransactionByServerId(long serverId);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long insert(Transaction transaction);

    @Update
    void update(Transaction transaction);

    @Delete
    void delete(Transaction transaction);

    @Query("DELETE FROM transactions")
    void deleteAll();

    @Query("SELECT * FROM transactions WHERE account_id = :accountId")
    List<Transaction> getTransactionsByAccountId(long accountId);

    @Query("SELECT * FROM transactions WHERE user_id = :userId")
    List<Transaction> getTransactionsByUserId(long userId);
} 

File: accounting_app/app/src/main/java/com/accounting/app/database/TransactionDao.kt
package com.accounting.app.database

import androidx.room.*
import com.accounting.app.models.Transaction

@Dao
interface TransactionDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertAll(transactions: List<Transaction>)

    @Query("SELECT * FROM transactions")
    suspend fun getAllTransactions(): List<Transaction>

    @Query("SELECT * FROM transactions WHERE account_id = :accountId")
    suspend fun getTransactionsForAccount(accountId: Long): List<Transaction>

    @Query("SELECT * FROM transactions WHERE date BETWEEN :fromDate AND :toDate")
    suspend fun getTransactionsBetweenDates(fromDate: Long, toDate: Long): List<Transaction>

    @Update
    suspend fun updateTransaction(transaction: Transaction)

    @Delete
    suspend fun deleteTransaction(transaction: Transaction)
} 